h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -118,7 +118,7 @@ public void failVerifyKnots(double[] knots) {
                 " manually");
         return;
       }
-      if (index > 0 && (knots[index]-knots[index-1]) <= 0) { // knots must be increasing in order
+      if (index > 0 && knots[index - 1] >= knots[index]) { // knots must be increasing in order
         error("knots formation", "knots not sorted in ascending order.");
```

Comment:
This order must be strictly increasing.  Hence, knots[index] > knots[index-1].

Reply:
Originally, it wasn't strict. I'll fix it.